packages/react/counter/src/Counter.tsx

```diff
@@ -89,11 +90,11 @@ export function Counter(props: CounterProps) {
         }
       )
 
-      const metricData = response.metricByName.counter
+      const metricData = response.counter
 
       setHasError(false)
 
-      return metricData.value
+      return metricData?.value ?? undefined
```

Comment:
I have a question regarding this.

If I'm not wrong, `metricData?.value` will:

- return `undefined` when `metricData` is `undefined`
- return `undefined` | `null` | `string` for `value

By using that nullish coalescing we are preventing to return `null`, we don't want to prevent this because `undefined` will show the error fallback while `null` is a expected value and it will show a `-` instead.

Reply:
Oh, very interesting. That is something we need to come back and write tests for. I will make the change for now. Is there any where else we need to look out for that?

Reply:
We just need to make sure to not disallow returning `null` for the Counter connected mode and to allow receiving it in props for the Static mode, but I think it is fine